exo-deploy/src/deployers/aws-deployer.ls

```diff
@@ -39,20 +41,59 @@ class AwsDeployer
 
   deploy: ->
     @terraform
-      ..get (err) ->
+      ..get (err) ~>
         | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting deploy to AWS"
-        ..apply (err) ->
-          | err => return process.stdout.write err.message
+        @_get-hosted-zone-id (hosted-zone-id) ~>
+          process.stdout.write "terraform starting deploy to AWS"
+          ..apply {hosted-zone-id}, (err) ->
+            | err => return process.stdout.write err.message
 
 
   nuke: ->
-    @terraform
-      ..get (err) ->
-        | err => return process.stdout.write err.message
-        process.stdout.write "terraform starting nuke from AWS"
-        ..destroy (err) ~>
+    @terraform-file-builder.generate-provider-credentials!
+    process.stdout.write "terraform starting nuke from AWS"
+    @_get-hosted-zone-id (hosted-zone-id) ~>
+      @terraform.destroy {hosted-zone-id}, (err) ~>
           | err => return process.stdout.write err.message
+          @_remove-hosted-zone @domain-name
+
+
+  _get-hosted-zone-id: (done) ->
+    @_hosted-zone-exists @domain-name, (err, id) ~>
+      | err       => return done process.stdout.write err.message
+      | id        => done id
+      | otherwise => @_create-hosted-zone @domain-name, (err, id) ->
+                        | err => return done process.stdout.write err.message
+                        done id
+
+
+  _hosted-zone-exists: (domain-name, done) ->
+    @route53 = new Aws.Route53 api-version: '2013-04-01'
+      ..list-hosted-zones null, (err, data) ~>
+        | err => return done process.stdout.write err.message
+        for hosted-zone in data.HostedZones when hosted-zone.Name is "#{domain-name}."
```

Comment:
This still doesn't express developer intent particularly well. What do you think about this:

```ls
return null, data.HostedZones |> find (.Name is "#{domain-name}.")